acq4/devices/FilterWheel/FilterWheel.py

```diff
@@ -0,0 +1,402 @@
+        with self.filterWheelLock:
+            if (self.currentFWPosition-1) not in self.filters:
+                return None
+            return self.filters[(self.currentFWPosition-1)]
+            #return self.objectives[self.currentSwitchPosition][selected]
+    
+    def listFilters(self):
+        """
+        Return a list of available filters.
+        """
+        with self.filterWheelLock:
+            return self.filters.values()
+    
+    def _allFilters(self):
+        ## used by (preferrably only) GUI interface
+        return self.filters
+    
+    
+    def deviceInterface(self, win):
+        return FilterWheelDevGui(self)
+
+
+class Filter(OptomechDevice):
+    
+    #class SignalProxyObject(QtCore.QObject):
+        #sigTransformChanged = QtCore.Signal(object) ## self
+    
+    def __init__(self, config, fw, key):
+        #self.__sigProxy = Objective.SignalProxyObject()
+        #self.sigTransformChanged = self.__sigProxy.sigTransformChanged
+        #self._config = config
+        self._config = config
+        self._fw = fw
+        self._key = key
+        #print config, key
+        key = str(key)
+        if key in config:
+            name = '%s-%s' %((int(key)+1), config[key]['name'])
+            self._description = config[key]['description']
+        else:
+            name = '%s-%s' %((int(key)+1), 'empty')
+            self._description = '-'
+        
+        OptomechDevice.__init__(self, fw.dm, {}, name)
+        
+        #if 'offset' in config:
+        #    self.setOffset(config['offset'])
+        #if 'scale' in config:
+        #    self.setScale(config['scale'])
+            
+    def key(self):
+        return self._key
+
+    def description(self):
+        return self._description
+    
+    def filterwheel(self):
+        return self._fw
+    
+    def __repr__(self):
+        return "<Filter %s.%s>" % (self._fw.name(), self.name())
+
+
+    
+class FilterWheelTask(DeviceTask):
+    #pass
+    # This is disabled--internal shutter in coherent laser should NOT be used by ACQ4; use a separate shutter.
+    #
+    def __init__(self, dev, cmd, parentTask):
+        DeviceTask.__init__(self, dev, cmd, parentTask)
+        self.dev = dev
+        self.cmd = cmd
+        self.parentTask = parentTask
+        #print parentTask
+        
+    def configure(self):
+        """Sets the state of a remote multiclamp to prepare for a program run."""
+        #print "mc configure"
+        with self.dev.filterWheelLock:
+            #self.state = self.dev.getLastState()
+            print 'command :', self.cmd
+            requiredPos = int(self.cmd['filterWheelPosition'][0]) # that the first character of string and convert it to int
+            if self.dev.currentFWPosition != requiredPos:
+                print 'current pos:', self.dev.currentFWPosition
+                print 'requir. pos:', requiredPos
+                self.dev.setPosition(requiredPos)
+            
+    def start(self):
```

Comment:
The method blocks until completion of the rotation. So the block you suggest is not necessary.